packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import type { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const MIGRATION_FILE_LOCATION = 'openzeppelin-cli-migration.json';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestList = await findManifests();
+  const { migrations, migrationData } = await migrateManifestFiles(manifestList);
+
+  for (const manifestFile in migrations) {
+    const newManifest = migrations[manifestFile];
+    await writeJSONFile(getNewManifestLocation(manifestFile), newManifest);
+  }
+
+  await writeJSONFile(MIGRATION_FILE_LOCATION, migrationData);
+}
+
+async function migrateManifestFiles(manifestList: string[]): Promise<MigrationOutput> {
+  const migrationData: Record<string, unknown> = {};
+  const migrations: Record<string, ManifestData> = {};
+
+  for (const manifestFile of manifestList) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+    const { manifestVersion } = oldManifest;
+
+    if (manifestVersion === undefined) {
+      continue;
```

Comment:
This looks like it should be an error. If there is no `manifestVersion` it's a really old CLI that didn't version its manifest files. We should fail if that's the case.